lib/mongoid/association/depending.rb

```diff
@@ -131,7 +131,9 @@ def _dependent_restrict_with_exception!(association)
 
       def _dependent_restrict_with_error!(association)
         if (relation = send(association.name)) && !relation.blank?
-          errors.add(association.name, RESTRICT_ERROR_MSG)
+          association_name_translated = ::I18n.translate("activerecord.models.#{association.name}", default: association.name.to_s)
```

Comment:
Why does this line reference `activerecord`?

I found https://stackoverflow.com/questions/18672861/rails-4-i18n-on-activerecord-attributes#23727347 which suggests that `human_attribute_name` should be used to translate attribute name, and it appears that Mongoid does support `human_attribute_name`.